pkg/fleetautoscalers/fleetautoscalers_test.go

```diff
@@ -2134,3 +2136,615 @@ func TestApplyListPolicy(t *testing.T) {
+					Buffer: &autoscalingv1.BufferPolicy{
+						BufferSize:  intstr.FromInt(1),
+						MinReplicas: 3,
+						MaxReplicas: 10,
+					},
+				},
+			},
+			want: expected{
+				replicas: 3,
+				limited:  false,
+				wantErr:  false,
+			},
+		},
+		"no end time": {
+			featureFlags: string(utilruntime.FeatureScheduledAutoscaler) + "=true",
+			now:          mustParseTime("2021-01-02T00:00:00Z"),
+			sp: &autoscalingv1.SchedulePolicy{
+				Between: autoscalingv1.Between{
+					Start: mustParseMetav1Time("2021-01-01T00:00:00Z"),
+				},
+				ActivePeriod: autoscalingv1.ActivePeriod{
+					Timezone:  "UTC",
+					StartCron: "* * * * *",
+					Duration:  "1h",
+				},
+				Policy: autoscalingv1.FleetAutoscalerPolicy{
+					Type: autoscalingv1.BufferPolicyType,
+					Buffer: &autoscalingv1.BufferPolicy{
+						BufferSize:  intstr.FromInt(1),
+						MinReplicas: 4,
+						MaxReplicas: 7,
+					},
+				},
+			},
+			want: expected{
+				replicas: 4,
+				limited:  true,
+				wantErr:  false,
+			},
+		},
+		"no cron time": {
+			featureFlags: string(utilruntime.FeatureScheduledAutoscaler) + "=true",
+			now:          mustParseTime("2021-01-01T0:30:00Z"),
+			sp: &autoscalingv1.SchedulePolicy{
+				Between: autoscalingv1.Between{
+					Start: mustParseMetav1Time("2021-01-01T00:00:00Z"),
+					End:   mustParseMetav1Time("2021-01-01T01:00:00Z"),
+				},
+				ActivePeriod: autoscalingv1.ActivePeriod{
+					Timezone: "UTC",
+					Duration: "1h",
+				},
+				Policy: autoscalingv1.FleetAutoscalerPolicy{
+					Type: autoscalingv1.BufferPolicyType,
+					Buffer: &autoscalingv1.BufferPolicy{
+						MinReplicas: 5,
+						MaxReplicas: 10,
+						BufferSize:  intstr.FromInt(1),
+					},
+				},
+			},
+			want: expected{
+				replicas: 5,
+				limited:  true,
+				wantErr:  false,
+			},
+		},
+		"no duration": {
+			featureFlags: string(utilruntime.FeatureScheduledAutoscaler) + "=true",
+			now:          mustParseTime("2021-01-01T0:30:00Z"),
+			sp: &autoscalingv1.SchedulePolicy{
+				Between: autoscalingv1.Between{
+					Start: mustParseMetav1Time("2021-01-01T00:00:00Z"),
+					End:   mustParseMetav1Time("2021-01-01T01:00:00Z"),
+				},
+				ActivePeriod: autoscalingv1.ActivePeriod{
+					Timezone:  "UTC",
+					StartCron: "* * * * *",
+				},
+				Policy: autoscalingv1.FleetAutoscalerPolicy{
+					Type: autoscalingv1.BufferPolicyType,
+					Buffer: &autoscalingv1.BufferPolicy{
+						BufferSize:  intstr.FromInt(8),
+						MinReplicas: 11,
+						MaxReplicas: 30,
+					},
+				},
+			},
+			want: expected{
+				replicas: 11,
+				limited:  true,
+				wantErr:  false,
+			},
+		},
+		"no start time, end time, cron time, duration": {
+			featureFlags: string(utilruntime.FeatureScheduledAutoscaler) + "=true",
+			now:          mustParseTime("2021-01-01T00:00:00Z"),
+			sp: &autoscalingv1.SchedulePolicy{
+				Policy: autoscalingv1.FleetAutoscalerPolicy{
+					Type: autoscalingv1.BufferPolicyType,
+					Buffer: &autoscalingv1.BufferPolicy{
```

Comment:
I found this difficult to read for a silly reason: the numbers kept changing. If I could make a suggestion here, just create a constant `BufferPolicy` at the start of the function, then use it throughout - you don't really need to test anything about the buffer policy except whether it's applied or not. Then here in `Buffer`, use the constant value, and in `want`, you can make two new constants: `expectedWhenActive` and `expectedWhenNotActive`, or something like that. Then the reader just sees:

```
  Buffer: buffer Policy
}
want: expectedWhenActive
```

Does that make sense?

Reply:
Yeah that makes sense.